### PR TITLE
[Serializer] Small CS improvement in ClassMetadata

### PR DESCRIPTION
**Pull request by hbgamra**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | N/A
| License       | MIT

CS improvement

### CODE REVIEW

**Comment by derrabus** [comment]
Thank you for your PR. However, we do not accept code-style changes on LTS branches, which is why I'm retargeting your PR to the 7.2 branch.

File: src/Symfony/Component/Serializer/Mapping/ClassMetadata.php [hunk 1]
```diff
@@ -78,11 +78,7 @@ public function merge(ClassMetadataInterface $classMetadata): void
 
     public function getReflectionClass(): \ReflectionClass
     {
-        if (!$this->reflClass) {
-            $this->reflClass = new \ReflectionClass($this->getName());
-        }
-
-        return $this->reflClass;
+        return $this->reflClass ??= new \ReflectionClass($this->getName());
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Serializer/Mapping/ClassMetadata.php:81]: I'd suggest one more change:
```
private \ReflectionClass $reflClass;
```

**Comment by stof** [comment]
Is this actually optimizing something ?

**Comment by nicolas-grekas** [comment]
I don't think so, this is just CS cleanup.

**Comment by stof** [comment]
then the PR description is misleading

File: src/Symfony/Component/Serializer/Mapping/ClassMetadata.php (hunk 1 above)
Comment (wouterj) [reply src/Symfony/Component/Serializer/Mapping/ClassMetadata.php:81]: I had a discussion about this recently in the static analysis channel of Symfony Slack. It is currently impossible to document (either natively or through PHPdocs) that a property is not initialized on construction. This makes using uninitialized properties for optional properties not a great choice for the readability and usability of the code.

Unless there are performance improvements of uninitialized properties vs nullable ones, I would be in favor to keep using nullable types to indicate that a property is optional.

Comment (nicolas-grekas) [reply src/Symfony/Component/Serializer/Mapping/ClassMetadata.php:81]: An other angle to this: keeping the property uinitialized makes it clearer that it shouldn't be accessed directly, whereas accepting `null` says: it's OK to deal with null values when accessing this property.
I'm more on the uninitialized side for this reason.

Comment (stof) [reply src/Symfony/Component/Serializer/Mapping/ClassMetadata.php:81]: there is no clear way to mark a property as being lazily initialized (either natively or in PHPdoc). so keeping a property uninitialized in the constructor actually does not make it clear that access it requires special care (if it is nullable, you have to take care of null, which might make you use the getter method that has a non-nullable return type).
Note that static analysis tools don't help there, as none of them deal with uninitialized properties correctly: either they pretend they don't exist or they forbid them.

**Comment by OskarStark** [comment]
I am 👎 on merging this, it is just syntax sugar to me

**Review by nicolas-grekas — APPROVED** [review]
We already merged the exact same change many times. We did it in a batch using some regexp but either we missed some occurrences, or new ones come up when merging from lower branches. Thus, this would bring consistency I think.
I updated the PR title+description.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @hbgamra.

**Pull request merged**